### PR TITLE
Create notebook 04: Conditionals

### LINKED ISSUE #5: Slides: Conditional Statements
Draw from:
- Intro class: [*Conditionals*](https://github.com/uc-python/intermediate-python-datasci/blob/master/notebooks/02-Conditionals.ipynb)
- Bana 6043: [*Lesson 6a*](https://github.com/bradleyboehmke/uc-bana-6043/blob/main/book/06-module/lesson-6a.ipynb)

### PR DESCRIPTION
**Pull request by eswan18**
This notebook ended up requiring minimal changes from the one in the intermediate class. I just took out a small section on vectorization and updated the "Applied Review" section.

### CODE REVIEW

File: notebooks/04-Conditionals.ipynb [hunk 1]
```diff
@@ -0,0 +1,1275 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "#### Conditionals"
```
Comment (augustopher) [inline notebooks/04-Conditionals.ipynb]:
```suggestion
    "### Conditionals"
```

File: notebooks/04-Conditionals.ipynb [hunk 2]
```diff
@@ -0,0 +1,1275 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "#### Iteration"
```
Comment (augustopher) [inline notebooks/04-Conditionals.ipynb]:
```suggestion
    "### Iteration"
```

File: notebooks/04-Conditionals.ipynb [hunk 3]
```diff
@@ -0,0 +1,1275 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "#### Form"
```
Comment (augustopher) [inline notebooks/04-Conditionals.ipynb]:
```suggestion
    "### Form"
```

File: notebooks/04-Conditionals.ipynb [hunk 4]
```diff
@@ -0,0 +1,1275 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "#### Execution Pattern"
```
Comment (augustopher) [inline notebooks/04-Conditionals.ipynb]:
```suggestion
    "### Execution Pattern"
```

File: notebooks/04-Conditionals.ipynb [hunk 5]
```diff
@@ -0,0 +1,1275 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "#### Why Use Conditionals"
```
Comment (augustopher) [inline notebooks/04-Conditionals.ipynb]:
```suggestion
    "### Why Use Conditionals"
```

Comment (augustopher) [reply notebooks/04-Conditionals.ipynb]: Actually, given the formatting for the other *Question* slides, you can probably just remove this header altogether.

File: notebooks/04-Conditionals.ipynb [hunk 6]
```diff
@@ -0,0 +1,1275 @@
...
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "slide"
+    }
+   },
+   "source": [
+    "### Non psuedo code"
```
Comment (augustopher) [inline notebooks/04-Conditionals.ipynb]: Go for the double hyphen!

File: notebooks/04-Conditionals.ipynb [hunk 7]
```diff
@@ -0,0 +1,1275 @@
...
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "slideshow": {
+     "slide_type": "fragment"
+    }
+   },
+   "source": [
+    "* You can include as many `elif` statements as you want\n",
+    "    * if you're familiar with a `switch` statement in other languages, this is Python's closest relative"
```
Comment (augustopher) [inline notebooks/04-Conditionals.ipynb:813]: Do we want comments (not on the slide) about the pattern matching available in 3.10? We are using it for the course, although many corporate environments might still be somewhere between 3.7 to 3.9.

**Review by augustopher — CHANGES_REQUESTED** [review]
Just some small `h4` updates.

File: notebooks/04-Conditionals.ipynb (hunk 6 above)
Comment (eswan18) [reply notebooks/04-Conditionals.ipynb]: It's always been a dream of mine

Comment (eswan18) [reply notebooks/04-Conditionals.ipynb]: Honestly I'm torn on which side should be single and which double

File: notebooks/04-Conditionals.ipynb (hunk 7 above)
Comment (eswan18) [reply notebooks/04-Conditionals.ipynb:813]: I didn't realize we were using it elsewhere. I'll plan to mention it in that case, and if I have time I'll come back and add a slide for it.

File: notebooks/04-Conditionals.ipynb (hunk 6 above)
Comment (eswan18) [reply notebooks/04-Conditionals.ipynb]: Somehow "psuedo" is misspelled and I never saw it until I opened the notebook json in Vim.

File: notebooks/04-Conditionals.ipynb (hunk 5 above)
Comment (eswan18) [reply notebooks/04-Conditionals.ipynb]: Removed in 011faf5

**Comment by eswan18** [comment]
Eh, broke something somehow, busy now but will fix later

File: notebooks/04-Conditionals.ipynb [hunk 8]
```diff
@@ -1216,13 +1205,14 @@
...
+   "execution_count": null,
    "metadata": {
     "slideshow": {
      "slide_type": "slide"
     }
    },
+   "outputs": [],
    "source": [
     "There's a lot of flexibility!"
    ]
```
Comment (augustopher) [inline notebooks/04-Conditionals.ipynb]: Looks like the markdown cell got change to code, and the string is not quoted when run.

**Comment by eswan18** [comment]
I have work events all day tomorrow and into the evening – @augustopher, if you approve this, can you just merge it in?

**Review by augustopher — APPROVED** [review]

**Pull request merged**